src/PolyType/Abstractions/AssociatedTypeRequirements.cs

```diff
@@ -0,0 +1,21 @@
+namespace PolyType.Abstractions;
+
+/// <summary>
+/// Describes the requirements for preparing an associated type.
+/// </summary>
+[Flags]
+public enum AssociatedTypeRequirements
+{
+    /// <summary>No requirements.</summary>
+    None = 0x0,
+
+    /// <summary>
+    /// A factory method should be prepared for the associated type.
```

Comment:
What alternative name to "factory" could we use to further emphasize that this specifically pertains to default constructors?